src/main/java/org/opentripplanner/routing/vertextype/TemporaryVertex.java

```diff
@@ -1,8 +1,26 @@
 package org.opentripplanner.routing.vertextype;
 
-/** Marker interface for temporary vertices */
+import org.opentripplanner.routing.graph.Vertex;
+
+/**
+ * Marker interface for temporary vertices.
+ * <p/>
+ * Remember to use the {@link #dispose(Vertex)} to delete the temporary vertex
+ * from the main graph after use.
+ */
 public interface TemporaryVertex {
-    public boolean isEndVertex();
+    boolean isEndVertex();
```

Comment:
I believe this method is meant to distinguish between temporary vertices at the start point and temporary vertices at the end point. Is that right? Does that mean all temporary vertices will be endpoints? I think that might be the case now, not sure if that will always be true.

Reply:
But in any case, rename or at least add a javadoc so it's clear what this method is.

Reply:
Currently yes, the temporary vertex/edge mechanism is only used for inserting endpoints in the graph. This should be made clear in the Javadoc, e.g. on the TemporaryEdge marker interface or the isTemporary() method you proposed as a replacement.

Reply:
I looked at it at it is not clear to me why we need the `isEndVertex()`. @abyrd feel free to add JavaDoc comment. Also, I did not change this, just took away the unnecessary `public` identifier :-) For the future, it would make sense to add a Temporary structure that is something else than just access/egress. The dispose is not using the `isEndVertex()` any more, so the only use of it is to check that the creation of the temporary graph is valid.